src/libcoretest/num/mod.rs

```diff
@@ -177,4 +177,61 @@ mod tests {
     test_impl_from! { test_u16i32, u16, i32 }
     test_impl_from! { test_u16i64, u16, i64 }
     test_impl_from! { test_u32i64, u32, i64 }
+
+    // Signed -> Float
+    test_impl_from! { test_i8f32, i8, f32 }
```

Comment:
I'll elaborate. I have a criterion for `Into`/`AsRef`: Suppose you have a function `fn f(arg: Into<U>)` (or `try!`, it also uses `From` for implicit conversions). Would you want it to implicitly accept `T`s? If yes, then implementing `Into<U>` for `T` is reasonable. All current implementations meet this criterion (except for `impl From<u32> for Ipv4Addr`). Would you want `fn f(arg: Into<f64>)` to accept `u8` implicitly? Probably not, there's a good chance this `u8` is not your desired number, but its index, for example. I.e. type safety prevents mistakes. On the other hand, would you want it to accept `u64` with explicit conversion? Probably yes, because zero error requirement is not normally important for such conversions.